packages/apollo-server-codegen/src/IR/Objects.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  FieldDefinitionNode,
+  FieldNode,
+  ObjectTypeDefinitionNode,
+  ObjectTypeExtensionNode,
+  OperationDefinitionNode,
+  parse,
+  StringValueNode,
+  isTypeExtensionNode,
+  InterfaceTypeDefinitionNode,
+  InterfaceTypeExtensionNode,
+  Kind
+} from "graphql";
+
+import { Translatable, Translator } from "../Translators";
+import { Description } from "./Descriptions";
+import { ResolverDefinition, TypelessResolverDefinition } from "./Resolvers";
+import { CompoundType, makeType, TypeDefinition } from "./Types";
+
+export const SELECTION_OFFSET = 7; // number of chars added in the below parse(`query { ${source} }`) call
```

Comment:
Can you expand on this some more in docs? I'm not sure I follow the need of it.

Reply:
Added more docs